ts/defs.bzl

```diff
@@ -331,60 +331,42 @@ def ts_project(
     typings_out_dir = declaration_dir if declaration_dir else out_dir
     tsbuildinfo_path = ts_build_info_file if ts_build_info_file else name + ".tsbuildinfo"
 
+    # Flags for what is emitted and which tool is emitting them
+    emit_js = not no_emit and (not emit_declaration_only)
+    emit_dts = not no_emit and (declaration or emit_declaration_only)
+    emit_tsc_js = emit_js and (not transpiler or transpiler == "tsc")
```

Comment:
I think it is easier to reason about if tsc is the fallback _if_ a custom transpiler is _not_ set:
```
emit_transpiler_js = emit_js and transpiler and transpiler != "tsc"
emit_transpiler_dts = emit_dts and declaration_transpiler
emit_tsc_js = emit_js and not emit_transpiler_js
emit_tsc_dts = emit_dts and not emit_transpiler_dts
```